src/uploaders.ts

```diff
@@ -33,26 +33,39 @@ export class XHRUploader extends AsyncUploader {
     public async upload(fetchPayload: fetchPayload): Promise<Response> {
         const response: Response = await this.makeRequest(
             this.url,
-            fetchPayload.body
+            fetchPayload.body,
+            fetchPayload.method as 'post' | 'get'
         );
         return response;
     }
 
-    private async makeRequest(url: string, data: string): Promise<Response> {
+    // XHR Ready States
+    // https://stackoverflow.com/questions/30522565/what-is-meaning-of-xhr-readystate-4#:~:text=State%204%20means%20that%20the,finished%20downloading%20the%20response%20content.
```

Comment:
I'd rather use this link...just seems more reputable
```suggestion
    // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState
```